server/api.js

```diff
@@ -551,6 +602,48 @@ router.postAsync("/match", ensure.isAdmin, async (req, res) => {
   res.send(match);
 });
 
+/**
+ * POST /api/warmup
+ * Submit a warmup
+ * Params:
+ *   - match: match ID
+ *   - playerNo: 1 - player 1, 2 - player 2
+ *   - warmup: the warmup map, could be beatmap link or ID
+ */
+router.postAsync("/warmup", async (req, res) => {
+  const match = await Match.findOne({ _id: req.body.match });
+  if (!(await canEditWarmup(req.user, req.body.playerNo, match))) {
+    logger.warn(`${req.user.username} tried to submit player ${req.body.playerNo} warmup for ${req.body.match}`);
+    return res.status(403).send("You don't have permission to do that");
+  }
+  try {
+    match[`warmup${req.body.playerNo}`] = await parseWarmup(req.body.warmup);
+  } catch (e) {
+    res.status(400).send(e.message);
+  }
+  console.log("User", req.user.username, "submitted warmup", req.body.warmup);
```

Comment:
nit: user `logger.info` instead of `console.log`

Reply:
Applied in 0035ab576aec7444588d93fd51845f5758c240f5